File: app/src/main/java/org/record/tiny/ui/MainActivity.java
package org.record.tiny.ui;

import android.content.Intent;
import android.os.Bundle;

import org.record.tiny.R;
import org.record.tiny.base.MvpActivity;
import org.record.tiny.component.CircleButton;
import org.record.tiny.ui.model.Article;
import org.record.tiny.ui.view.MainPresenter;
import org.record.tiny.ui.view.MainView;

import java.util.List;

import butterknife.Bind;
import butterknife.OnClick;

@SuppressWarnings("Allw")
public class MainActivity extends MvpActivity<MainPresenter> implements MainView {

    @Bind(R.id.bt_edit)
    CircleButton mEditButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @OnClick(R.id.bt_edit)
    void goEditActivity() {
        startActivity(new Intent(this, EditActivity.class));
    }

    @Override
    protected MainPresenter createPresenter() {
        return new MainPresenter();
    }

    @Override
    public void getYear(String year) {

    }

    @Override
    public void getMonth(String month) {

    }

    @Override
    public void getDay(String day) {

    }

    @Override
    public void getContext(List<Article> articles) {

    }

    @Override
    public void getLocal(String local) {

    }

    @Override
    public void error(int error) {

    }
}

File: app/src/main/java/org/tiny/record/view/MainActivity.kt
package org.tiny.record.view

import android.os.Bundle
import org.tiny.component.ComponentManager
import org.tiny.component.IMain
import org.tiny.lib.core.BaseActivity
import org.tiny.lib.core.BasePresenter
import org.tiny.lib.core.BaseView
import org.tiny.record.R
import org.tiny.record.databinding.ActivityMainBinding

class MainActivity : BaseActivity<ActivityMainBinding, BasePresenter<BaseView>>() {
    override fun createLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun createPresenter(): BasePresenter<BaseView>? {
        return BasePresenter(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val main: IMain = ComponentManager.getInstance().getComponent(IMain::class.java)
        mBinding!!.model = TestViewModel("" + main.getTestString())

    }
}
